### ISSUE TITLE
FireFox V81 Breaks multi line tabs

### ISSUE THREAD

**Issue by davidvinograd** [MrOtherGuy/firefox-csshacks#77]
Hi,

Just installed FF V81 and lost the multi-line tabs feature. Silly me, next time I will not be so quick

Tried to revert to a prior firefox version and but can't use the current profile so all (book marks, extensions ..) are lost and will need to fix this.

As a stop-gap while the fix is coming, how  do I see / access beyond the first N displayed tabs?

Thanks,

Dave

**Comment by MrOtherGuy** [comment]
The latest version of multi-row_tabs.css should work just fine. If you update to latest and it *still* doesn' t work then I wanna know about it.

(There was another issue opened about this very topic just yesterday but I closed it because the latest version indeed solved the problem)

**Comment by davidvinograd** [comment]
I have updated the file from github and restarted FF, but the problem 
appears to continue. Can you suggest what else I can try?

I will be away at hospital for an eye injection for the next couple of 
hours so will not respond till after 6PM UK time.

Thanks for your help.

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by MrOtherGuy** [comment]
Hmm..

First thing I would want you to confirm is if it works all by itself. I mean, remove all your other custom CSS and see of it works or not.

Also, please tell what OS you have just in case it's somehow related.

**Comment by tastyratz** [comment]
V81 broke multi row tabs for me as well. I was able to see multiple rows and remain scrolling however only a single row displayed at a time. Replacing the CSS file contents with what's in the git here was the fix even though 81 just came out and it was modified 2 months ago. 

If rows aren't still present and scrollable I'd say check if it's the right profile and the userchrome file has the correct filename too - ESPECIALLY if you were using a different CSS file before.

**Comment by tobe** [comment]
Same issue, tried the same solution as @tastyratz (replace whatever you have in the `userChrome.css` with what's in the repo, eventhough it's 2m old), works just fine now.

Very odd.

**Comment by patfla** [comment]
Simply copying over worked for me three.  It appears to me that the FF 81 install made a new profile and while the new profile did contain a chrome/userChrome.css the problem had something to do with the new profile.

**Comment by MrOtherGuy** [comment]
For anyone wondering why it works even if its 2months old. Well, 81 development cycle started over 2 months ago and I have probably changed whatever needed to be changed back when 81 was still in Nightly phase.

**Comment by SanderBouwhuis** [comment]
@MrOtherGuy
I use 3 of your scripts and all seems to work (now that I updated them) in Fx81.

https://github.com/MrOtherGuy/firefox-csshacks/tree/master/chrome/window_control__placeholder_support.css
https://github.com/MrOtherGuy/firefox-csshacks/tree/master/chrome/tabs_on_bottom.css
https://github.com/MrOtherGuy/firefox-csshacks/tree/master/chrome/multi-row_tabs.css

Thank you, thank you, thank you for your continued efforts. It is HIGHLY appreciated!

**Comment by ChieftainY2k** [comment]
It seems that v81 changed the userChrome.css , I've just overwritten it with fresh content from multi-row_tabs.css and it works again :)

**Comment by MrOtherGuy** [comment]
> It seems that v81 installator changed the userChrome.css , I've just overwritten it with the fresh content from multi-row_tabs.css and it works again :)

I think there is something else going on. Firefox is not supposed to ever write to userChrome.css file. Only thing it ever does is load it on startup.

**Comment by davidvinograd** [comment]
Hi,

The OS is win 10 1903 64bit.

Unlike others on this thread I continue to have no success.

I downloaded the zip file, stopped FF, replaced all file in my profile 
with those from the zip file and restarted FF - no change. Deleted all 
files except multi-row-tabs, stopped and started FF - no change.

I still can not move beyond the first row of tabs. There is a white 
solid blob on the far right.

Any help would be appreciated.

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by davidvinograd** [comment]
Further, in the instructions write up there is a reference to prefs.js 
being in the profile directory. This is not true in my case.

And finally, in the display of about:profiles, the root directory is in 
the sub tree AppDate\Roaming, while the local directory is in the sub 
tree AppData\Local. I have been making changes to files in Local. Is 
that correct?

Thanks,

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------
From: "David Vinograd" <david.vinograd@gmail.com>
To: "MrOtherGuy/firefox-csshacks" 
<reply@reply.github.com>
Sent: 25/09/2020 10:57:33
Subject: Re[2]: [MrOtherGuy/firefox-csshacks] FireFox V81 Breaks multi 
line tabs (#77)

**Comment by MrOtherGuy** [comment]
I think your userChrome file may just be setup incorrectly. It needs to be under that roaming folder. Local only has cache and basically stuff that you can safely delete if you wish.

If you go to about:support and use the open profile folder button there, yhen it should lead to the roaming folder which should have among other things prefs.js

You would need to create the chrome folder into that folder and inside the chrome folder a file - userChrome.css

Then you can use the userChrome.css file to import the multirow css- or you can copy the contents of the multirow file directly into userChrome.css

**Comment by patfla** [comment]
Is there a way to figure out if userChrome.css loaded successfully during FF startup?

**Comment by MrOtherGuy** [comment]
You can open Browser Toolbox and from there style editor tab and find out if it lists userChrome.css. But it's often easier to teat with some userChrome.css that will definitely do something and if it has any effect then you know it's loaded.

So test with something like `#nav-bar{ display:none !important }`

 If the main toolbar is visible the Firefox is not loading userChrome.css. But if the toolbar is not visible then Firefox is indeed loading it.

**Comment by patfla** [comment]
Thanks - that's very helpful.

**Comment by davidvinograd** [comment]
Hi,

I replicated zip file files in the chrome directory in the roaming 
subtree and restarted FF, as you suggested

The display is just plain weird. The command bar is missing unless I 
move the cursor over top left of the screen. The book marks tool bar 
icons are spread out rather than being tight together.
There is only a single row of tabs. But there is a  down arrow on the 
far right which shows other tabs but only the icon is visible - the rest 
is very hard to read (black text on grey background but only when cursor 
highlighted. My addons are there but only visible when the cursor is 
over them.

I realize this may be taking too much of your time but I do need and 
appreciate your help in sorting out this mess.

Thanks,

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by patfla** [comment]
I'm not a regular on this project but have been using the extension for a while and am a former developer and reasonably literate in that regard.

Personally, I"d like to see the first, say, 20 lines of the userChrome.css you're using to make sure you have the right code.

Also - you've gotten your current profile location from about:profiles - correct?  In particular, I have a number of profile directories listed there - one needs to use the directory that's marked with "in use".  That is, the current, active profile directory.

I apologize to you or anyone else if what I suggest is stupidly obvious and I'll happily desist if that's the case.

**Comment by davidvinograd** [comment]
Hi,

Further, if I comment out all the import directives from the sample 
userchrome.css file (I use the example files and just rename them and 
add an import directive for the multi tabs) the screen is still badly 
laid out. Disabling all the addons has not impact.

Can anyone suggest anything to try?

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------
From: "David Vinograd" <david.vinograd@gmail.com>
To: "MrOtherGuy/firefox-csshacks" 
<reply@reply.github.com>
Sent: 26/09/2020 15:07:19
Subject: Re[2]: [MrOtherGuy/firefox-csshacks] FireFox V81 Breaks multi 
line tabs (#77)

**Comment by davidvinograd** [comment]
Hi,

As noted the userChrome.css is the one that is shipped. I rename the 
examples. It is attached. Note the mods - there is only one active 
import file, multi-tabs.

I believe I have the correct profile dir.

As a final check I rename the chrome dir. A rebooted FF looks ok - no 
visual funnies. Except the multiple tabs appear on one line that must be 
scrolled - not very usable.

Any help is appreciated.

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by patfla** [comment]
Pare it down to a minimum for a start.   Clean everything in your local FF profile out first.

Then, considering just multi-row_tabs (it's all I want - at least at the moment), put the following one line into your userChrome.css ($PROFILE\chrome\userChrome.css)

@import url(chrome/multi-row_tabs.css);

next, in a subdirectory chrome (yes chrome/chrome - $PROFILE\chrome\chrome\) put the one file, multi-row_tabs.css, from here in the github repo.  It's in a chrome directory on github as well.

Restart FF.

If successful, later you can add other modules as needs be.

**Comment by MrOtherGuy** [comment]
> It is attached. Note the mods - there is only one active
import file, multi-tabs.

Neither of your messages have any attachments. I don't think email attachments are added to the github issues, if that's what you have been doing.

But like @patfla said, start with simple things. Don't try to use any of the files from the repository, not the examples or anything else. Instead, try with just some simple rule in userChrome.css and try to replicate some expected result.

So just add this to userChrome.css `#nav-bar{ display:none !important }` 

The result should be that the main toolbar is non-existent. If the result is anything else then there is something else weird going on.

**Comment by davidvinograd** [comment]
Following your advice and that of patfa I created a userChrome.css with 
a single line importing mult-tabs - to no gain. When I added the nav bar 
directive the screen was quite blank - no controls, no active extensions 
etc. Taking that line out put the display right but I had to restore the 
tabs from a tabs saver extension.

Note that I used a separate directory to import the multi-tabs file 
from.

What do you suggest next?

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by MrOtherGuy** [comment]
> ...When I added the nav bar
directive the screen was quite blank - no controls, no active extensions

That sounds about right. Okay, so next do NOT try any import directive. Instead copy the contents of the multi-row_tabs.css directly into userChrome.css

**Comment by davidvinograd** [comment]
Success!!!!!

Many Many thanks!!!

But why??

And how to fix it?

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by MrOtherGuy** [comment]
Awesome. I don't know why or how, but I suspect that you have been using `@import` statements wrong or imported wrong or too many files or something like that.

**Comment by davidvinograd** [comment]
Many thanks to all for their help in resolving this.

But I am still confused about what is going on.

If I embed the text of mult-row_tabs.css in userChrome.css all is fine. 

If I try to import multi-row_tabs.css from the chrome or an other directory (both inferior to the main chrome directory) no joy.

Can anyone suggest where to look or what experiments to try?

Thanks,

David

**Comment by MrOtherGuy** [comment]
Ehat **exactly** does your userChrome.css contain when you try to use imports?

**Comment by patfla** [comment]
Another thing you might want to try is to produce a new FF profile at least or maybe installation.  You might not want to uninstall your present FF installation but you can download a zip file of FF and extract it onto disk making a new FF instance.  

Close out your standard FF and run this one and it'll create a new profile where you can experiment.  In particular, I'd first add the one line for multi-row_tabs to the new userChrome.cs (creating whatever necessary directories along the way) and then put multi-row_tabs.css down in the chrome/chrome directory.  Make sure that this all works in the way in which it should work so as to allow you to install other modules as well.

Did this tell you what was wrong with the old profile?  No, but it could take a lot of poking around to figure out what was wrong with the old one - and you now have a new one that works as it should.  And in the event that you want to switch to the new one?  That's left as an exercise to the reader.

There's not time enough in life to completely understand everything goes wrong with every piece of software people use.

**Comment by davidvinograd** [comment]
Not sure how to answer that. userChrome.css contains the import 
statement. If some temp file is created I am not aware of it. The 
browser behaves as if multi-tabs is not loaded.

The text of userChrome.css is as follows.

/* Example userChrome.css file */



/* Import your desired components first */
/* My adds */

@import url  ("chrome/multi-row_tabs.css") ;


/* #nav-bar{ display:none !important }*/

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by davidvinograd** [comment]
Patfla,

Not sure what that would solve. The new profile would be missing the 
environment I need to work - bookmarks, tabs, etc. If import worked I 
could not migrate to it. If import did not work I am no further in 
solving it.

I need to explore the current profile and see , if I can, what may be 
the cause.

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------

**Comment by MrOtherGuy** [comment]
> The text of userChrome.css is as follows.
> ...

Okay, so with that you need to make sure your file/folder structure is as follows:

    <profile_folder>
      |_<...>
      |_chrome
      |    |_userChrome.css
      |    |_chrome
      |    |    |_multi-row_tabs.css
      |    |_content
      |_<...>
      |_prefs.js
      |_places.sqlite
      |_whole-bunch-of-other-files&folders

Also, there must not be a space between `url` and the parentheses - this might be real reason why your imports don't work.

**Comment by davidvinograd** [comment]
Well now it works!!!

I checked the directory tree structure and no changes. I checked the 
file userChrome.css and all looked good, but just to be sure I retyped 
the import line and now all is well. Perhaps a non printing character 
got inserted?

Wow, what a saga.

Many, Many, Many thanks

David

--------------------------------------------------------------------------------------------------------

David Vinograd                     Email: david.vinograd@gmail.com

Flat 8                             Home Page: 
davidvinograd.000webhostapp.com

128 Richmond Hill                  8 Grassy Path

Richmond                           PeaksIsland

TW10 6RN UK                        Portland, ME04108US

UK: 44-(0)20-8-940-5723            US: 1-(1)207-766-5840



------ Original Message ------
